### PR TITLE
adding mongodb storage backend

### LINKED ISSUE #79: mongodb storage backend
I would like to add support for a mongodb storage back-end, and I'm also willing to contribute the code. I've tested it locally, with tests passing and benchmarks appearing favorable. Are you open to such a contribution? Cheers.

### PR DESCRIPTION
**Pull request by f00actual**
adding mongodb storage backend

closes #79

### CODE REVIEW

**Comment by majst01** [comment]
Nice so far, please check the output of `golangci-lint run -p bugs -p unused`

File: testing_test.go [hunk 1]
```diff
@@ -3,13 +3,13 @@ package ipam
...
 	"fmt"
-	"os"
-	"sync"
-	"testing"
-
 	"github.com/testcontainers/testcontainers-go"
 	"github.com/testcontainers/testcontainers-go/wait"
 	clientv3 "go.etcd.io/etcd/client/v3"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"os"
```
Comment (majst01) [inline testing_test.go:10]: please keep stdlibrary imports at the top

**Review by majst01 — CHANGES_REQUESTED** [review]

**Comment by majst01** [comment]
Can you please add some useful entries to the Readme as well, especially the table with the supported mongodb versions and some "normalized" benchmark numbers. According to the ci run the benchmark shows a ~ 2x performance of the postgres version.

**Comment by f00actual** [comment]
What approach did you take for the op/sec normalization in the README.md? It appears the latest bench results ran on different hardware and they might need updated overall. What do you think?

**Comment by majst01** [comment]
> What approach did you take for the op/sec normalization in the README.md? It appears the latest bench results ran on different hardware and they might need updated overall. What do you think?

Like your current layout, but the problem persist, these hardware specs are hard to achieve. And what is actually running in the ci pipeline is not known.

**Review by majst01 — APPROVED** [review]

**Comment by majst01** [comment]
I think this is ready for merging

**Pull request merged**